lib/components/artwork_grids/infinite_scroll_grid.js

```diff
@@ -30,16 +31,47 @@ const PageEndThreshold = 1000
  *   - see ARMasonryCollectionViewLayout for details on how to deal with last works sticking out
  *   - the calculation currently only takes into account the size of the image, not if e.g. the sale message is present
  */
+
+type Props = {
+  /** The direction for the grid, currently only 'column' is supported . */
+  sectionDirection: string;
+
+  /** The arity of the number of sections (e.g. columns) to show */
+  sectionCount: number;
+
+  /** The inset margin for the whole grid */
+  sectionMargin: number;
+
+  /** The per-item margin */
+  itemMargin: number;
+
+  /** All the artworks for the grid */
+  artworks: any[];
+
+  /** A non-optional callback for create per-page request state. */
+  queryForState: () => any;
+
+  /** A non-optional callback to generate the GraphQL query. */
+  queryForPage: (component: InfiniteScrollArtworksGrid, page: number, state: any) => string;
+
+  /** A callback that is called once all artworks have been queried. */
+  onComplete?: () => void;
+
+  /** When you get the results from the GraphQL, under what keypath can the artworks be found, applied via `_.get()`  */
```

Comment:
it has sentience

![Giphy](http://media3.giphy.com/media/wmqAxgvojFKj6/giphy.gif)